funcx_endpoint/funcx_endpoint/executors/high_throughput/funcx_manager.py

```diff
@@ -308,6 +324,59 @@ def pull_tasks(self, kill_event):
                     kill_event.set()
                     break
 
+                elif type(message) == tuple and message[0] == 'TASK_CANCEL':
+                    self.task_cancel_lock.acquire()
+                    task_id = message[1]
+                    logger.info(f"Received TASK_CANCEL request for task: {task_id}")
+                    if task_id not in self.task_worker_map:
+                        logger.warning(f"Task:{task_id} is not in task_worker_map.")
+                        logger.warning("Possible duplicate cancel or race-condition")
+                        continue
+                    # Cancel task by killing the worker it is on
+                    worker_id_raw = self.task_worker_map[task_id]['worker_id']
+                    worker_to_kill = self.task_worker_map[task_id]['worker_id'].decode('utf-8')
+                    worker_type = self.task_worker_map[task_id]['task_type']
+                    logger.debug(f"Cancelling task running on worker:{self.task_worker_map[task_id]}")
+                    try:
+                        logger.info(f"Removing worker:{worker_id_raw} from map")
+                        self.worker_map.start_remove_worker(worker_type)
+                        self.worker_map.remove_worker(worker_id_raw)
+                        logger.info(f"Popping worker:{worker_to_kill} from worker_procs")
+                        proc = self.worker_procs.pop(worker_to_kill)
+                        logger.warning(f"Sending process:{proc.pid} terminate signal")
+                        proc.terminate()
+                        try:
+                            proc.wait(1)  # Wait 1 second before attempting SIGKILL
+                        except subprocess.TimeoutExpired:
+                            logger.exception("Process did not terminate in 1 second")
+                            logger.warning(f"Sending process:{proc.pid} kill signal")
+                            proc.kill()
+                        else:
+                            logger.debug(f"Worker process exited with : {proc.returncode}")
+
+                        raise TaskCancelled(worker_to_kill, self.uid)
+                    except Exception as e:
+                        logger.exception(f"Raise exception, handling: {e}")
+                        result_package = {'task_id': task_id,
+                                          'container_id': worker_type,
```

Comment:
I'm very confused in reading funcx code in general about what is the difference between container_id and a worker_type. are they the same thing? (what docker calls an image id?)